test/hw_tests/hwtest_timestamp_standalone.py

```diff
@@ -0,0 +1,153 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from collections import namedtuple
+from pathlib import Path
+import re
+import subprocess
+from threading import Thread
+import time
+import unittest
+import contextlib
+from typing import Sequence
+
+
+UDP_DATA_HEX_KEYS = ["scan_counter", "from_theta"]
+UDP_DATA_PORT = "55115"
+WAIT_TIME_S = 30.0
+
+UdpData = namedtuple("UdpData", UDP_DATA_HEX_KEYS + ["stamp"])
+TimeStamp = namedtuple("TimeStamp", ["stamp", "scan_counter"])
+Pair = namedtuple("Pair", ["first_frame", "last_frame", "callback"])
+
+
+def convert_hex_string(hex_string: str):
+    return int.from_bytes(bytes.fromhex(hex_string), "little")
+
+
+def read_hex_value(line: str, key: str):
+    return convert_hex_string(re.search(r"%s: ([0-9a-f]+)" % key, line).group(1))
+
+
+def read_number(line: str, key: str):
+    return float(re.search(r"%s: ([0-9.]+)" % key, line).group(1))
+
+
+def read_udp_line(line: str):
+    """ Return empty list if one of the keys is not found.
+    """
+    try:
+        return UdpData(*[read_hex_value(line, key) for key in UDP_DATA_HEX_KEYS],
+                       read_number(line, "time_epoch") * 1000000000)
+    except AttributeError:
+        return []
+
+
+def read_udp_pakets(data: str):
+    """ Lines which do not contain data for all keys are omitted.
+    """
+    return [read_udp_line(line) for line in data.splitlines()]
+
+
+def read_timestamp_output(data: str):
+    stamps = []
+    for l in data.splitlines():
+        with contextlib.suppress(ValueError):
+            stamps.append(read_timestamp(l))
+    return stamps
+
+
+def read_timestamp(line: str):
+    l = line.split(": ")
+    return TimeStamp(int(l[2]), int(l[3]))
+
+
+def run_command(command, timeout, **kwargs):
+    print("\n\nExecuting: %s\n" % (command))
+    start = time.time()
+    full_output = ""
+    with subprocess.Popen(command, stdout=subprocess.PIPE, shell=True, **kwargs) as process:
+        while (time.time() - start) < timeout:
+            output = process.stdout.readline().decode()
+            if output:
+                full_output += output
+        process.kill()
+    return full_output.strip()
+
+
+def pair_udp_data_and_stamps(udp_data: Sequence[UdpData], stamps: Sequence[TimeStamp]):
+    pairs = []
+    for t in stamps:
+        first = None
+        last = None
+        for u in udp_data:
```

Comment:
```suggestion
        for u in filter(t.scan_counter == u.scan_counter, udp_data):
```
should work? But first see the comment below about ordering `pairs`.

Reply:
or maybe we could even generate a list of all frames and use `min`/`max` operators.